pkg/platform/kube/platform.go

```diff
@@ -1730,6 +1722,39 @@ func (p *Platform) enrichAndValidateFunctionConfig(ctx context.Context, function
 	return nil
 }
 
+// enrichFunctionNodeSelector enriches function node selector
+// if node selector is not specified in function config, we firstly try to get it from project CRD
+// if it is missing in project CRD, then we try to get it from platform config
+func (p *Platform) enrichFunctionNodeSelector(ctx context.Context, functionConfig *functionconfig.Config) error {
+	if functionConfig.Spec.NodeSelector == nil {
+		functionProject, err := p.Platform.GetFunctionProject(ctx, functionConfig)
+		if err != nil {
```

Comment:
I think what needs to happen here is a merge, not condition.

imagine you have 
```
env default label select [a=b, p=w]
platform label selector [a=c, e=f]
function label selector [g=d, a=a]
```
the result should be - function label selector = [g=d, a=a, e=f, p=w]
if user dont want the label inherit by project, perhaps we can allow him to set its value to "" which means, drop it all together, e.g.:

```
env default label select [a=b, p=w]
platform label selector [a=c, e=f]
function label selector [g=d, a=a, p=]
```
the result should be - function label selector = [g=d, a=a, e=f]

Reply:
@liranbg agreed, it makes much more sense, will rework